### PR TITLE
add documentation for grpc

### PR DESCRIPTION
**Pull request by jeveleth**

### CODE REVIEW

File: docs/guides/running-grpc.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+## Installation
```
Comment (hbagdi) [inline docs/guides/running-grpc.md]: Let's rename this to `using-ingress-with-grpc.md`. 
Also, please add in a link to this guide in the `README.md` of this directory.

File: docs/guides/running-grpc.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+This is expected as Kong does not yet know how to proxy the request.
+
+#### Running GRPC
+
+
+1. Add a grpc deployment and service
+
+`kubectl apply -f ../../deploy/manifests/sample-apps/grpc.yaml`
```
Comment (hbagdi) [inline docs/guides/running-grpc.md]: These need to be absolute Github links. We do not expect that the user will clone this repository.

Comment (jeveleth) [reply docs/guides/running-grpc.md]: Any objection to my using bit.ly links to point to aspirational paths: e.g., https://raw.githubusercontent.com/Kong/kubernetes-ingress-controller/master/deploy/manifests/sample-apps/sample-grpc-ingress.yaml?

File: docs/references/annotations.md [hunk 3]
```diff
@@ -129,6 +131,16 @@ Please follow the
...
 
+## `configuration.konghq.com/protocol`
+
+This annotation can set a protocol � `http`, `https`, `grpc`, or `grpcs` �
+on a Service resource.
+
+## `configuration.konghq.com/protocols`
+
+This annotation can set a pair of protocols (`http`,`https`) or (`grpc`,`grpcs`)
+on an Ingress resource.
```
[the remark this answers is not in the document]
Comment (jeveleth) [reply docs/references/annotations.md]: I've updated to add some background about Routes and Services, but let me know if that misses the point.

File: deploy/manifests/sample-apps/grpc.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: grpcbin
+  labels:
+    app: grpcbin
+spec:
+  ports:
+  - name: http
```
Comment (hbagdi) [inline deploy/manifests/sample-apps/grpc.yaml]: nit: s/http/grpc

File: docs/guides/README.md [hunk 5]
```diff
@@ -26,21 +26,23 @@ Kong Ingress controller:
...
+- [Using Redis for rate-limiting](redis-rate-limiting.md)
   This guide walks through how to use Redis for storing rate-limit information
   in a multi-node Kong deployment.
-- [Integrate Kong Ingress Controller with Prometheus/Grafana](prometheus-grafana.md)  
+- [Integrate Kong Ingress Controller with Prometheus/Grafana](prometheus-grafana.md)
   This guide walks through the steps of how to deploy Kong Ingress Controller
   and Prometheus to obtain metrics for the traffic flowing into your
   Kubernetes cluster.
-- [Configuring circuit-breaker and health-checking](configuring-health-checks.md)  
+- [Configuring circuit-breaker and health-checking](configuring-health-checks.md)
```
Comment (hbagdi) [inline docs/guides/README.md]: These spaces are intentional. They add a `\n` to the markdown rendering. If you see this vs master markdown rendering, you will see that the description of the guide is now is not on a newline anymore. Please revert this change.

File: docs/references/annotations.md [hunk 6]
```diff
@@ -129,6 +131,16 @@ Please follow the
...
+## `configuration.konghq.com/protocol`
+
+This annotation can set a protocol � `http`, `https`, `grpc`, or `grpcs` �
+on a Service resource. The services you define will map to the [Kong Service](https://docs.konghq.com/1.4.x/admin-api/#service-object), operating on the protocol that you set.
+
+## `configuration.konghq.com/protocols`
+
+This annotation can set a pair of protocols (`http`,`https`) or (`grpc`,`grpcs`)
+on an Ingress resource. The paths in the Ingress resources you define will map to the [Kong Routes](https://docs.konghq.com/1.4.x/admin-api/#route-object), operating on the protocol that you set.
+
```
Comment (hbagdi) [inline docs/references/annotations.md]: Let's try to keep these lines under 80 columns limit.
Two changes:
1. Can you change the link to `latest` instead of  `1.4.x` so that it is automatically kept up to date?
2. Can we also add in as to what it means? For `protocol`, I'd add that: "This sets the protocol to be used for communication between Kong and the service, internally in the k8s cluster.", and similar for `protocols`, I'd say that it is the protocol to be used for communication between the external user/service and the ingress point (kong).

File: deploy/manifests/sample-apps/grpc.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: grpcbin
+  labels:
+    app: grpcbin
+spec:
+  ports:
+  - name: grpc
+    port: 80
```
Comment (hbagdi) [inline deploy/manifests/sample-apps/grpc.yaml]: The Ingress rule points to 9001 and it works probably because Kong sends traffic directly to the pod and doesn't care about this port. But this should be 9001 as well or the ingress rule should be 80.

Comment (hbagdi) [reply deploy/manifests/sample-apps/grpc.yaml]: And apologies for not catching in the earlier reviews. This is the last change we need. Otherwise this is good to go.

Comment (jeveleth) [reply deploy/manifests/sample-apps/grpc.yaml]: Done

**Comment by hbagdi** [comment]
Thank you for getting this across the finish line!
